### DISCUSSION TITLE
Trying to use serialport@6 in my electron app, but i'm getting an error.

### DISCUSSION THREAD

**Discussion by lucas-gama** [serialport/node-serialport#2682] [Q&A] [answered]
Versions:

**node**: 8.17.0 (32 bits version)
**electron**: 1.3.2
**enviroment**: Windows 11 x64
**serialport**: 6.2.2

After installing serialport@6 successfully, i'm importing the lib with nodeRequire (library [requirejs](https://github.com/requirejs/requirejs)), the same way i do with another dependencies of my project:

`var SerialPort =  nodeRequire('serialport');`

But i'm getting the following error:

`Error: A dynamic link library (DLL) initialization routine failed.
\\?\C:\Users\Lucas\AppData\Local\Programs\PDV Desktop\1.0.0\resources\app\node_modules\serialport\build\Release\serialport.node
    at Error (native)
    at process.module.(anonymous function) [as dlopen] (ELECTRON_ASAR.js:168:20)
    at Object.Module._extensions..node (module.js:583:18)
    at Object.module.(anonymous function) [as .node] (ELECTRON_ASAR.js:168:20)
    at Module.load (module.js:473:32)
    at tryModuleLoad (module.js:432:12)
    at Function.Module._load (module.js:424:3)
    at Module.require (module.js:483:17)
    at require (internal/module.js:20:19)
    at bindings (C:\Users\Lucas\AppData\Local\Programs\PDV Desktop\1.0.0\resources\app\node_modules\serialport\node_modules\bindings\bindings.js:81:44)`

Does anybody know what may be happening? Thanks.

**Comment by GazHank** [comment]
Is there a reason why you are using such old unsupported versions, are you not able to use modern versions of node, electron and serialport?

**Reply by lucas-gama** [reply]
Unfortunatelty the project that i'm working on is a legacy app, and it would cost an insane amount of time refactoring it to make it possible  updating libs and node for newer versions...

**Reply by GazHank** [reply, marked as answer]
yikes, I've not tried using these old versions so am not sure if they will still work on windows 11; and the oldest documentation still available is for v7 :-(

Am I correct to think that you have an old electron 1.3.2 app that didn't include serialport, and you are looking to retrofit serialport into it rather than an existing app where serialport has recently stopped working?

Also, are you using the prebuild binaries or built from source?  (per https://serialport.io/docs/7.x.x/guide-installation)

Assuming that running these old version is still feasible, then it could be an issue of incapability between the versions.

Node 8.17 is node ABI version 57 (per https://nodejs.org/en/download/releases)
Electron 1.3.2 is node ABI 49 (per https://github.com/lgeiger/electron-abi)

You can find the compatibility between SerialPort and the node version @ https://serialport.io/docs/guide-platform-support

It is normally necessary to compile serialport to target the ABI of electron rather than the ABI of your node version.  Since electron 1.3.2 is aligned with node 6 rather than 8, this could be part of the problem, if prebuild binaries or used or if the compile is targeting the wrong abi
